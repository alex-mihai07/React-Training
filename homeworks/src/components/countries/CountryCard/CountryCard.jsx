import React from "react";
import styles from './CountryCard.module.css'
import PropTypes from "prop-types";

const CountryCard = (props) => {
    const {name, population, region, capital, flagUrl} = props

    return (
            <div className={styles.box}>
                <img src={flagUrl}/>
                <div>
                    <h2>{name}</h2>
                    <p><span className={styles.bold}>Population:</span> {population}</p>
                    <p><span className={styles.bold}>Region:</span> {region}</p>
                    <p><span className={styles.bold}>Capital:</span> {capital}</p>
                </div>
            </div>
    )
}

CountryCard.propTypes = {
    name: PropTypes.string,
    population: PropTypes.number,
    region: PropTypes.string,
    capital: PropTypes.string,
    flagUrl: PropTypes.string,
}

export default CountryCard