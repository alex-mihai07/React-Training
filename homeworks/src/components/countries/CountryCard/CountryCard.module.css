.box{
    width: 240px;
    height: 360px;

    -webkit-box-shadow: 1px 1px 5px 7px rgba(179,177,179,0.11);
    -moz-box-shadow: 1px 1px 5px 7px rgba(179,177,179,0.11);
    box-shadow: 1px 1px 5px 7px rgba(179,177,179,0.11);
    border-radius: 3px;
    border:2px grey;
    float: left;
    margin: 30px;
}

.bold{
    font-weight: bold;
}

img{
    width: 240px;
    height: 130px;
}