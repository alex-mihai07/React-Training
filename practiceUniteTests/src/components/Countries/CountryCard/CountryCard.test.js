import React from 'react'
import {render} from '@testing-library/react'
import CountryCard from "./CountryCard";
import {countriesMock} from "../../../utils/tests/global-mocks"


const mockedCountryCardData = countriesMock[0]

describe("Country cards component", () => {
    it("should match snapshot", () => {
        const {asFragment} = render(<CountryCard data={mockedCountryCardData}/>)

        expect(asFragment()).toMatchSnapshot();
    })
})
