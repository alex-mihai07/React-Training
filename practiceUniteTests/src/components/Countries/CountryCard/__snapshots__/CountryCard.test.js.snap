// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country cards component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <img
      alt="Mock name"
      class="cardImg"
      src="https://restcountries.eu/data/ala.svg"
    />
    <strong
      class="countryName"
    >
      Mock name
    </strong>
    <div
      class="details"
    >
      <div>
        <strong>
          Population:
        </strong>
         100
      </div>
      <div>
        <strong>
          Region:
        </strong>
         Mock region
      </div>
      <div>
        <strong>
          Capital:
        </strong>
         Mock capital
      </div>
    </div>
  </div>
</DocumentFragment>
`;
